drivers/ethernet/eth_stm32_hal.c

```diff
@@ -930,7 +992,13 @@ static int eth_initialize(const struct device *dev)
 	HAL_ETH_DMARxDescListInit(heth, dma_rx_desc_tab,
 		&dma_rx_buffer[0][0], ETH_RXBUFNB);
 
-#endif /* !CONFIG_ETH_STM32_HAL_API_V1 */
+#elif defined(CONFIG_ETH_STM32_HAL_API_V2)
+	/* This function requires the Ethernet interface to be
+	 * properly initialized. In auto-negotiation mode, it reads the speed
+	 * and duplex settings to configure the driver accordingly.
+	 */
```

Comment:
remove this comment, as it is wrong, since we switched to using the phy api completly